### ISSUE TITLE
LLDB target.import-std-module causes false ambiguity errors

### ISSUE THREAD

**Issue by rnikander** [llvm/llvm-project#57867]
Example:

```cpp
#include <fstream>
#include <iostream>

int main() {
  std::ifstream fout("/tmp/junk");
  bool b = fout.is_open();
  std::cout << "is_open? " << b << std::endl;
  return 0;
}
```

Following the answer [here](https://stackoverflow.com/questions/70445481/error-couldnt-lookup-symbols-when-calling-an-stl-method-in-lldb) about having standard library symbols available when debugging, I add this line to `~/.lldbinit`:

```
settings set target.import-std-module true
```

Then compile and debug...

```console
% clang++ -std=c++20 -g -O0 amg2.cc
% lldb a.out                       
(lldb) target create "a.out"
Current executable set to '/Users/rob/Workspace/BugsAndQuestions/LLDB/a.out' (x86_64).
(lldb) break set --file amg2.cc --line 8
Breakpoint 1: where = a.out`main + 192 at amg2.cc:8:3, address = 0x0000000100002d50
(lldb) run
Process 1774 launched: '/Users/.../a.out' (x86_64)
is_open? 0
Process 1774 stopped
* thread #1, queue = 'com.apple.main-thread', stop reason = breakpoint 1.1
    frame #0: 0x0000000100002d50 a.out`main at amg2.cc:8:3
   5   	  std::ifstream fout("/tmp/junk");
   6   	  bool b = fout.is_open();
   7   	  std::cout << "is_open? " << b << std::endl;
-> 8   	  return 0;
   9   	}
Target 0: (a.out) stopped.
(lldb) po fout.is_open()
error: expression failed to parse:
error: <user expression 0>:1:6: call to member function 'is_open' is ambiguous
fout.is_open()
~~~~~^~~~~~~
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/c++/v1/fstream:1169:10: candidate function
    bool is_open() const;
         ^

note: candidate function

(lldb) 
```

Versions:

```console
% clang --version
Apple clang version 14.0.0 (clang-1400.0.29.102)
Target: x86_64-apple-darwin22.1.0
Thread model: posix
InstalledDir: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin
% lldb --version
lldb-1400.0.30.3
Apple Swift version 5.7 (swiftlang-5.7.0.127.4 clang-1400.0.29.50)
```

**Comment by Michael137** [comment]
FYI doesn't reproduce for me on top-of-tree (tested with RelDebInfo 1c62af3e23cab41074f7ce0ba86a93bea82b99b9 without assertions). Closing for now, re-open if you're still seeing this (or similar) on latest lldb
